File: test2/src/main/java/xyz/yggdrazil/delaunay/voronoi/Edge.java
package xyz.yggdrazil.delaunay.voronoi;

import xyz.yggdrazil.delaunay.geom.Point;

/**
 * Edge.java
 *
 * @author Connor
 */
public class Edge {

    public int index;
    public Center d0, d1;  // Delaunay edge
    public Corner v0, v1;  // Voronoi edge
    public Point midpoint;  // halfway between v0,v1
    public int river;

    public void setVornoi(Corner v0, Corner v1) {
        this.v0 = v0;
        this.v1 = v1;
        midpoint = new Point((v0.loc.getX() + v1.loc.getX()) / 2, (v0.loc.getY() + v1.loc.getY()) / 2);
    }
}

File: test2/src/main/java/xyz/yggdrazil/delaunay/voronoi/Edge.kt
package xyz.yggdrazil.delaunay.voronoi

import xyz.yggdrazil.delaunay.geom.Point

/**
 * @author Connor
 */
class Edge {

    var index: Int = 0
    var d0: Center? = null
    var d1: Center? = null  // Delaunay edge
    var v0: Corner? = null
    var v1: Corner? = null  // Voronoi edge
    var midpoint: Point? = null  // halfway between v0,v1
    var river: Int = 0

    fun setVornoi(v0: Corner, v1: Corner) {
        this.v0 = v0
        this.v1 = v1
        midpoint = Point((v0.loc!!.x + v1.loc!!.x) / 2, (v0.loc!!.y + v1.loc!!.y) / 2)
    }
}
